static/js/components/forms/CheckoutForm.js

```diff
@@ -190,7 +190,7 @@ export class InnerCheckoutForm extends React.Component<InnerProps, InnerState> {
       values.couponCode &&
       values.couponCode !== "" &&
       coupon &&
-      coupon.code === values.couponCode
+      coupon.code === values.couponCode.trim().replace(/\s\s+/g, " ")
```

Comment:
I guess you should only use `trim()` and not the `replace(...)` because we're only concerned about the whitespace on either end, not in the middle. Basically we shouldn't be modifying what the user has entered apart from any whitespace that has been added by mistake around the start/end.